Lib/unittest/case.py

```diff
@@ -857,22 +857,28 @@ def assertAlmostEqual(self, first, second, places=None, msg=None,
             raise TypeError("specify delta or places not both")
 
         if delta is not None:
-            if abs(first - second) <= delta:
+            diff = abs(first - second)
+            if diff <= delta:
                 return
 
-            standardMsg = '%s != %s within %s delta' % (safe_repr(first),
-                                                        safe_repr(second),
-                                                        safe_repr(delta))
+            standardMsg = '%s != %s within %s delta (%s difference)' % (
+                safe_repr(first),
+                safe_repr(second),
+                safe_repr(delta),
+                safe_repr(diff))
         else:
             if places is None:
                 places = 7
 
-            if round(abs(second-first), places) == 0:
+            diff = abs(second - first)
+            if round(diff, places) == 0:
                 return
 
-            standardMsg = '%s != %s within %r places' % (safe_repr(first),
-                                                          safe_repr(second),
-                                                          places)
+            standardMsg = '%s != %s within %r places (%s difference)' % (
```

Comment:
Maybe I'm understanding the code wrong but `assertNotAlmostEqual` fails if the two numbers are almost equal and just "off" by a little. So in case of:

```
assertNotAlmostEqual(1.00000001, 1.0, places=7)
```

The failure will look like this:

````
AssertionError 1.00000001 == 1.0 within 7 places
```

...which is already enough. Makes sense?